### PR TITLE
Security Manager Fixes

### PR DESCRIPTION
**Pull request by jamezp**
The first commit reverts https://github.com/jboss/jboss-jaxrs-api_spec/commit/b578e70dd99a427758df7991c6456e40298d1dd0, see https://issues.redhat.com/browse/JBEE-185

The second commit adds some new privileged actions https://issues.redhat.com/browse/JBEE-253.

### CODE REVIEW

**Review by asoldano — APPROVED** [review]

File: jaxrs-api/src/main/java/javax/ws/rs/client/ClientBuilder.java [hunk 1]
```diff
@@ -72,30 +72,29 @@ public static ClientBuilder newBuilder() {
...
-                    loader = pClass.getClassLoader();
+                    ClassLoader loader = pClass.getClassLoader();
                     if (loader == null) {
                         loader = ClassLoader.getSystemClassLoader();
                     }
-
+                    URL targetTypeURL = loader.getResource(classnameAsResource);
+                    errorMessage = "ClassCastException: attempting to cast"
+                            + delegate.getClass().getClassLoader().getResource(classnameAsResource)
+                            + " to " + targetTypeURL;
```
Comment (rsearls) [inline jaxrs-api/src/main/java/javax/ws/rs/client/ClientBuilder.java:84]: These lines and lines 92-94 are duplicates.  Can't this be consolidated into a single point of definition?

**Comment by rsearls** [comment]
These 3 classes have a lot of common code.   Can the common code be refactored into a utiltiy class that all can use?

jaxrs-api/src/main/java/javax/ws/rs/ext/FactoryFinder.java 
jaxrs-api/src/main/java/javax/ws/rs/client/FactoryFinder.java 
jaxrs-api/src/main/java/javax/ws/rs/sse/FactoryFinder.java

File: jaxrs-api/src/main/java/javax/ws/rs/client/ClientBuilder.java (hunk 1 above)
Comment (jamezp) [reply jaxrs-api/src/main/java/javax/ws/rs/client/ClientBuilder.java:84]: I thought about it. Really it's just the `delegate.getClass().getClassLoader()` that needs to be privileged, but there didn't seem to be a good way to break that up without two privileged blocks so I did it like this.

**Comment by jamezp** [comment]
I don't think there is since it uses privileged actions. If there was a single class then anyone could use it to load class with the permissions required. It could lead to security issues.

**Pull request merged**